tutorials/snapcraft/snap-a-nodejs-app.md

```diff
@@ -0,0 +1,666 @@
+
+
+
+```bash
+$ sudo systemctl start snap.chuck-norris-webserver.node-service
+$ systemctl status -l snap.chuck-norris-webserver.node-service
+● snap.chuck-norris-webserver.node-service.service - Service for snap application chuck-norris-webserver.node-service
+   Loaded: loaded (/etc/systemd/system/snap.chuck-norris-webserver.node-service.service; enabled; vendor preset: enabled)
+   Active: active (running) since mar. 2016-11-15 15:12:42 CET; 2s ago
+ Main PID: 15622 (node)
+    Tasks: 5
+   Memory: 42.1M
+      CPU: 431ms
+   CGroup: /system.slice/snap.chuck-norris-webserver.node-service.service
+           └─15622 node /snap/chuck-norris-webserver/x1/bin/chuck-norris-app
+
+nov. 15 15:12:42 tidus systemd[1]: Started Service for snap application chuck-norris-webserver.node-service.
+nov. 15 15:12:43 tidus snap[15622]: Server listening on: http://localhost:80
+```
+Of course, there is a `systemctl restart` command as well!
+Live log following:
+
+
+```bash
+$ sudo journalctl -fu snap.chuck-norris-webserver.node-service
+-- Logs begin at mer. 2016-01-13 14:59:21 CET. --
+nov. 15 14:40:11 tidus snap[14410]: Server listening on: http://localhost:80
+nov. 15 14:47:35 tidus snap[14410]: Quoted http://api.chucknorris.io/jokes/ijpnw0birv-ogszti_wuag
+nov. 15 15:03:11 tidus systemd[1]: Stopping Service for snap application chuck-norris-webserver.node-service...
+nov. 15 15:03:11 tidus systemd[1]: Stopped Service for snap application chuck-norris-webserver.node-service.
+nov. 15 15:12:42 tidus systemd[1]: Started Service for snap application chuck-norris-webserver.node-service.
+nov. 15 15:12:43 tidus snap[15622]: Server listening on: http://localhost:80
+nov. 15 15:16:14 tidus snap[15622]: Quoted http://api.chucknorris.io/jokes/gll-vunmsluzrad8sfafwg
+nov. 15 15:16:18 tidus snap[15622]: Quoted http://api.chucknorris.io/jokes/qwv-rz6xtjmhiayprtgnvw
+nov. 15 15:16:20 tidus snap[15622]: Quoted http://api.chucknorris.io/jokes/ag_6paerrkg-mxfjjqw4ba
+nov. 15 15:17:53 tidus snap[15622]: Quoted http://api.chucknorris.io/jokes/elgv2wkvt8ioag6xywykbq
+```
+You can omit -f to not following the logs live (meaning, new inputs are printed as they come): it will only prints the messages that were logged when running the command.
+
+
+```bash
+Small tip: sometimes, multiple services coordinates to provide a functionality, or a service impacts some system services like network-manager. To get all logs and follow them live, just don’t provide the -u option with the service name. For instance, journalctl -f will provide you all logs, lively printed, from your system, including your services.
+```
+
+
+## That’s all folks!
+Duration: 1:00
+
+### There is no Esc key on Chuck Norris’ keyboard, because no one escapes Chuck Norris.
+
+Congratulations! As Chuck, you made it!
+
+
+positive
+: Final code 
+Your final code directory should now look like [this]. Do not hesitate to download and build your snap from it if you only read it through!
```

Comment:
[this] -> [this](https://github.com/ubuntu/snap-tutorials-code/tree/master/build-a-nodejs-service/final)